engine/src/test/java/com/datatorrent/stram/client/AppPackageTest.java

```diff
@@ -95,22 +97,51 @@ public void testAppPackage() throws Exception
     Assert.assertEquals(System.getProperty("apex.version", "3.4.0"), json.getString("dtEngineVersion"));
     Assert.assertEquals("lib/*.jar", json.getJSONArray("classPath").getString(0));
 
-    JSONObject application1 = json.getJSONArray("applications").getJSONObject(0);
-    JSONObject application2 = json.getJSONArray("applications").getJSONObject(1);
+    // Test if there are fixed number of applications
+    Assert.assertEquals("There are two applications", 2, json.getJSONArray("applications").length());
 
     Map<String, JSONObject> apps = new HashMap<>();
+    for (int i = 0; i < json.getJSONArray("applications").length(); i++) {
+      apps.put(json.getJSONArray("applications").getJSONObject(i).getString("name"), json.getJSONArray("applications").getJSONObject(i));
+    }
 
-    apps.put(application1.getString("name"), application1);
-    apps.put(application2.getString("name"), application2);
-
+    // Testing if expected applications are found
     Assert.assertEquals(true, apps.containsKey("MyFirstApplication"));
     Assert.assertEquals(true, apps.containsKey("MySecondApplication"));
 
-    Assert.assertEquals("mydtapp-1.0-SNAPSHOT.jar", apps.get("MyFirstApplication").getString("file"));
+    // Retrieve applications found
+    JSONObject myFirstApplication = apps.get("MyFirstApplication");
+    JSONObject mySecondApplication = apps.get("MySecondApplication");
+
+    if (myFirstApplication != null) {
+
+      Assert.assertEquals("mydtapp-1.0-SNAPSHOT.jar", myFirstApplication.getString("file"));
+
+      String errorStackTrace = myFirstApplication.getString("errorStackTrace");
+      Assert.assertTrue("There should be no error stack trace", errorStackTrace.equals("null"));
+      Assert.assertNotNull(apps.get("MyFirstApplication").get("dag"));
+      JSONObject dag = myFirstApplication.getJSONObject("dag");
+      Assert.assertTrue("There is at least one stream", dag.getJSONArray("streams").length() >= 1);
```

Comment:
Should it assert on exact number of streams? Is number of streams in myFirstApplication unknown?